distributed/http/proxy.py

```diff
@@ -41,6 +53,8 @@ async def http_get(self, port, host, proxied_path):
             return await self.proxy(port, proxied_path)
 
         async def open(self, port, host, proxied_path):
+            if host not in self.host_allowlist:
+                self.host_allowlist.append(host)
```

Comment:
Should this be tested as well? (same for the change in lines 41-42)

Reply:
I'm not sure how to do this w/o some obvious mocking/assert calls as all our tests are ran on the same host, but okay. I'll look at that as well.

Reply:
Okay, small adjustment to a test to verify that bit of code. Note that basically none of `GlobalProxyHandler` is directly tested but thru integration type testing so followed suit there. The test (`test_dashboard_non_standard_ports`) will fail now w/o GlobalProxyHandler updating `host_allowlist`. https://github.com/dask/distributed/pull/8528/commits/f9cd3dc2cc672da09071872a0b047f9e82924e06

Reply:
Thanks! 🚀

Reply:
It looks like this is still not tested, but given that testing appears to be tricky, I consider this non-blocking.

Reply:
Ya, the _second_ one there, is only hit on a websocket connection unfortunately.